go/def.bzl

```diff
@@ -999,9 +999,172 @@ def cgo_library(name, srcs,
       **kwargs
   )
 
+################
+
+def _go_repository_impl(ctx):
+  fetch_repo = ctx.path(ctx.attr._fetch_repo)
+
+  if ctx.attr.commit and ctx.attr.tag:
+    fail("cannot specify both of commit and tag", "commit")
+  if ctx.attr.commit:
+    rev = ctx.attr.commit
+  elif ctx.attr.tag:
+    rev = ctx.attr.tag
+  else:
+    fail("neither commit or tag is specified", "commit")
+
+  result = ctx.execute([
+      fetch_repo,
+      '--dest', ctx.path(''),
+      '--remote', ctx.attr.importpath,
+      '--rev', rev])
+  if result.return_code:
+    fail("failed to fetch %s: %s" % (ctx.attr.importpath, result.stderr))
+
+def _new_go_repository_impl(ctx):
+  _go_repository_impl(ctx)
+  gazelle = ctx.path(ctx.attr._gazelle)
+
+  result = ctx.execute([
+      gazelle,
+      '--go_prefix', ctx.attr.importpath, '--mode', 'fix',
+      ctx.path('')])
+  if result.return_code:
+    fail("failed to generate BUILD files for %s: %s" % (
+        ctx.attr.importpath, result.stderr))
+
+_go_repository_attrs = {
+    "importpath": attr.string(mandatory = True),
+    "commit": attr.string(),
+    "tag": attr.string(),
+
+    "_fetch_repo": attr.label(
+        default = Label("@io_bazel_rules_go_repository_tools//:bin/fetch_repo"),
+        allow_files = True,
+        single_file = True,
+    ),
+}
+
+go_repository = repository_rule(
+    _go_repository_impl,
+    attrs = _go_repository_attrs,
+)
+
+new_go_repository = repository_rule(
+    _new_go_repository_impl,
+    attrs = _go_repository_attrs + {
+        "_gazelle": attr.label(
+            default = Label("@io_bazel_rules_go_repository_tools//:bin/gazelle"),
+            allow_files = True,
+            single_file = True,
```

Comment:
Also,
executable = True
or is that not used in WORKSPACE rules the same as BUILD rules?